pandas/tests/frame/test_convert_to.py

```diff
@@ -175,7 +172,71 @@ def test_to_records_index_name(self):
     def test_to_records_with_unicode_index(self):
         # GH13172
         # unicode_literals conflict with to_records
-        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a')\
+        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a') \
             .to_records()
         expected = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
         tm.assert_almost_equal(result, expected)
+
+    def test_to_records_with_tz_gmt(self):
+        """
+        This test verifies the bug id #13937
+        """
+        data = [datetime.datetime.now(pytz.timezone('UTC')) for _ in range(10)]
+        df = DataFrame({'datetime': data})
+        df.set_index('datetime', inplace=True)
+        dfgmt = df.tz_convert("GMT")
+        tm.assertIsNot(df.to_records()['datetime'][0].tzinfo,
+                       dfgmt.to_records()['datetime'][0].tzinfo)
+
+    def test_to_records_with_tz_convert_to_utc(self):
+        """
+        This test verifies the bug id #13937
+        """
+        data_utc = [datetime.datetime.now(
+            pytz.timezone('UTC')) for _ in range(10)]
+        df = DataFrame({'datetime': data_utc})
+        data_gmt = [datetime.datetime.now(
+            pytz.timezone('GMT')) for _ in range(10)]
+        df2 = DataFrame({'datetime': data_gmt})
+        df.set_index('datetime', inplace=True)
+        df2.set_index('datetime', inplace=True)
+        dfgmt = df2.tz_convert("UTC")
+        tm.assertIs(df.to_records()['datetime'][0].tzinfo,
+                    dfgmt.to_records()['datetime'][0].tzinfo)
+
+    def test_to_records_with_tz_as_equal_frame(self):
+        """
+        This test verifies the bug id #13937
+        """
+        dt1 = [datetime.datetime(2011, 8, 15, 8, 15, 12, i + 1)
+               for i in range(10)]
+        dt2 = [datetime.datetime(2011, 8, 15, 8, 15, 12, i + 1, pytz.UTC)
+               for i in range(10)]
+        dt3 = [pytz.utc.localize(dt1[i]) for i in range(10)]
+
+        df1 = DataFrame({'datetime': dt2})
+        df2 = df1.set_index('datetime')
+
+        df3 = DataFrame({'datetime': dt3})
+        df4 = df3.set_index('datetime')
+
+        tm.assert_frame_equal(df2, df4)
```

Comment:
Your question is not fully clear to me. 
If you want to test `to_records`, you can also assert that the resulting array is equal (with `assert_numpy_array_equal` instead of `assert_frame_equal`)